tests/unit/lms/services/oauth_http_test.py

```diff
@@ -225,6 +232,33 @@ def test_refresh_access_token_raises_ValidationError_if_the_response_is_invalid(
                 sentinel.token_url, sentinel.redirect_uri, sentinel.auth
             )
 
+    def test_refresh_access_token_raises_ConflictError_on_concurrent_refresh(
+        self,
+        svc,
+        oauth2_token_service,
+    ):
+        oauth2_token_service.try_lock_for_refresh.side_effect = TryLockError()
+
+        with pytest.raises(ConflictError):
+            svc.refresh_access_token(
+                sentinel.token_url, sentinel.redirect_uri, sentinel.auth
+            )
```

Comment:
I think ideally you need to also assert that it chained the exception properly, and doesn't swallow the root exception:

```suggestion
        oauth2_token_service.try_lock_for_refresh.side_effect = TryLockError()

        with pytest.raises(ConflictError) as exc_info:
            svc.refresh_access_token(
                sentinel.token_url, sentinel.redirect_uri, sentinel.auth
            )
            
        assert exc_info.value.__cause__ == oauth2_token_service.try_lock_for_refresh.side_effect
```

But if `OAuthHTTPService` just lets the `TryLockError` fly through (rather than catching it and re-raising it as `ConflictError`) then I think arguably you don't actually need this test at all. You _could_ add a test that "if `try_lock_for_refresh()` raises `TryLockError` then `refresh_access_token()` raises the same `TryLockError`" but I think you could also argue for simply omitting the test.